src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/ClientServices.cs

```diff
@@ -24,32 +25,217 @@ namespace Microsoft.Azure.IIoT.OpcUa.Protocol.Services {
+                    }
+                    finally {
+                        trustedStore.Close();
+                    }
+                }
+                catch (Exception ex) {
+                    _logger.Warning(ex, $"Certificate Setter failed to open the trusted peer store.");
+                }
+
+                _OpcApplicationConfig.CertificateValidator.Update(_OpcApplicationConfig.SecurityConfiguration);
+            }
+        }
 
         /// <summary>
         /// Create client host services
         /// </summary>
         /// <param name="logger"></param>
         /// <param name="maxOpTimeout"></param>
-        public ClientServices(ILogger logger, TimeSpan? maxOpTimeout = null) {
+        public ClientServices(ILogger logger, TimeSpan? maxOpTimeout = null){
+
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _maxOpTimeout = maxOpTimeout;
+            _configuration = new ClientServicesConfig(null);
+
             // Create discovery config and client certificate
+            _OpcApplicationConfig = CreateApplicationConfiguration(
+                TimeSpan.FromMinutes(2), TimeSpan.FromMinutes(2));
+            InitApplicationSecurityAsync().Wait();
+
+            _timer = new Timer(_ => OnTimer(), null, kEvictionCheck, Timeout.InfiniteTimeSpan);
+
+        }
+
+
+        /// <summary>
+        /// Create client host services
+        /// </summary>
+        /// <param name="logger"></param>
+        /// <param name="configuration"></param>
+        /// <param name="maxOpTimeout"></param>
+        public ClientServices(ILogger logger,
+            IClientServicesConfig configuration,
+            TimeSpan? maxOpTimeout = null){
+
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _configuration = configuration ?? 
+                throw new ArgumentNullException(nameof(configuration)); ;
             _maxOpTimeout = maxOpTimeout;
-            _config = CreateApplicationConfiguration(TimeSpan.FromMinutes(2),
-                TimeSpan.FromMinutes(2));
-            Certificate = CertificateFactory.CreateCertificate(
-                _config.SecurityConfiguration.ApplicationCertificate.StoreType,
-                _config.SecurityConfiguration.ApplicationCertificate.StorePath, null,
-                _config.ApplicationUri, _config.ApplicationName,
-                _config.SecurityConfiguration.ApplicationCertificate.SubjectName, null,
-                CertificateFactory.defaultKeySize, DateTime.UtcNow - TimeSpan.FromDays(1),
-                CertificateFactory.defaultLifeTime, CertificateFactory.defaultHashSize,
-                false, null, null);
-            _timer = new Timer(_ => OnTimer(), null, kEvictionCheck,
-                Timeout.InfiniteTimeSpan);
+
+            // Create discovery config and client certificate
+            _OpcApplicationConfig = CreateApplicationConfiguration(
```

Comment:
for naming convention, lower case the first letter after the underscore.  I would even remove the opc, just _applicationConfig maybe.